library/core/src/panic/location.rs

```diff
@@ -122,12 +133,43 @@ impl<'a> Location<'a> {
     ///
     /// panic!("Normal panic");
     /// ```
+    ///
+    /// # C/C++ compatibility
+    ///
+    /// Although `file` returns an `&str`, the characters of `file` are guaranteed to be followed
+    /// by a nul-terminator. This allows for greater interoperabilty with C and C++ code using
+    /// `__FILE__` or `std::source_location::file_name`, both of which return nul-terminated
+    /// `const char*`.
```

Comment:
Note that under Stacked Borrows, it is Undefined Behavior to access the trailing null byte since it is outside the range of this reference. So it seems ill-advised to write documentation that encourages people to perform such accesses.

Reply:
Removed. At one point I actually had a test for this and removed it when I realized this issue, but I forgot to remove the docs. Thanks for catching!